### DISCUSSION TITLE
Accessor messing with FileUpload

### DISCUSSION THREAD

**Discussion by davmarek** [filamentphp/filament#13407] [Help]
### Package

Form builder

### Package Version

v3.2.90

### How can we help you?

TL;DR
Filament FileUpload doesn't add the `storage/` part of the path to DB. I want to have a default image to be shown on frontend when the value of the column is `null`. Using Accessor to add `storage/` to the path breaks the Filament FileUpload.

---
I have a model Event.php with a column for an image path. The stored images should be at `storage/app/public/img/event/xxx.jpg` (or accessible from `public/storage/img/event/xxx.jpg`)

**The problem is that I want to have a default image** that I render in case no image is defined and the column value is null. I have defined an accessor so that I could use the asset() helper function on frontend:
```php
protected function image(): Attribute
{
    return Attribute::make(
        get: fn (?string $value) => Storage::url($value) ?? self::DEFAULT_IMAGE,
    );
}
```
DEFAULT_IMAGE value is `img/default/event.jpg` so it points to this path `public/img/default/event.jpg`

Filament stores only `img/event/xxx.jpg` in DB so I need to add the `storage/` path prefix somewhere.... (currently trying to do it in the accessor, other option is to add it on frontend but how to have the default image defined there without too much hardcoding?).

BUT Filament probably adds the `storage/` to the path behind the scenes when displaying the image preview. SO the accessor adds the `storage/` part and Filament adds it again. (that's my theory)

What is the best practice here? Is there a way to tell Filament to store the image path WITH the `storage/` part? Should I do it some completely other way?
